Library/Homebrew/language/python.rb

```diff
@@ -111,7 +111,8 @@ def detected_python_shebang(formula = self)
           raise ShebangDetectionError.new("Python", "formula has multiple Python dependencies")
         end
 
-        python_shebang_rewrite_info(Formula[python_deps.first].opt_bin/"python3")
+        python_dep = python_deps.first
+        python_shebang_rewrite_info(Formula[python_dep].opt_bin/python_dep.sub("@", ""))
```

Comment:
Should we need to remove `python` in `ALLOWED_USES_FROM_MACOS_DEPS`?

https://github.com/Homebrew/brew/blob/4a623e0a2532a00a42474c755b8c03e927594676/Library/Homebrew/rubocops/uses_from_macos.rb#L78-L94

Reply:
Maybe. Though I suspect `uses_from_macos "python"` should probably just get `#!/usr/bin/env python3`, since the alias could change without the formulae that reference it be rebuilt.

It's probably also typically used when the formula doesn't really care about the specific Python version.

Reply:
Actually, trying to use `detected_python_shebang` with `uses_from_macos "python"` likely produces an error on macOS at the moment, no?

Reply:
```ruby
class Httpstat < Formula
  include Language::Python::Shebang

  desc "Curl statistics made simple"
  homepage "https://github.com/reorx/httpstat"
  url "https://github.com/reorx/httpstat/archive/1.3.1.tar.gz"
  sha256 "7bfaa0428fe806ad4a68fc2db0aedf378f2e259d53f879372835af4ef14a6d41"
  license "MIT"
  revision 2

  bottle do
    sha256 cellar: :any_skip_relocation, all: "219cfdfb663f1f260322ba637902065e5cad3559eeb2e338c198875f3cfd4b15"
  end

  uses_from_macos "curl"
  uses_from_macos "python"

  def install
    if OS.linux? || MacOS.version >= :catalina
      rw_info = python_shebang_rewrite_info("/usr/bin/env python3")
      rewrite_shebang rw_info, "httpstat.py"
    end
    bin.install "httpstat.py" => "httpstat"
  end

  test do
    assert_match "HTTP", shell_output("#{bin}/httpstat https://github.com")
  end
end
```

On Linux:

```console
$ brew irb          
==> Interactive Homebrew Shell
Example commands available with: `brew irb --examples`
irb(main):001:0> f = Formula["httpstat"]
=> #<Formula httpstat (stable) /home/PanXuehai/Projects/brew/Library/Taps/homebrew/homebrew-core/Formula/httpstat.rb>
irb(main):002:0> f.deps
=> #<Dependencies: [#<Dependency: "curl" []>, #<Dependency: "python" []>]>
irb(main):003:0> 
```

On macOS:

```console
$ brew irb          
==> Interactive Homebrew Shell
Example commands available with: `brew irb --examples`
irb(main):001:0> f = Formula["httpstat"]
=> #<Formula httpstat (stable) /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/httpstat.rb>
irb(main):002:0> f.deps
=> #<Dependencies: []>
irb(main):003:0> 
```

Reply:
On Linux, got `"python"` (not resolved to `"python@3.9"`) in `deps`. Got empty `deps` on macOS.

Reply:
Yes, pretty sure trying to use `detected_python_shebang` errors with `uses_from_macos "python"`, and using `#!/usr/bin/env python3` is probably more appropriate for it. The `httpstat` example proves my point, I think.